opg-base-1604/Dockerfile

```diff
@@ -1,79 +1,54 @@
-
-# Application user and location
-RUN  useradd -m -d /app app && \
-     mkdir -p /var/log/app /data /var/run/app && chown app /var/log/app /data /var/run/app
-
-ADD  docker/service/beaver /etc/sv/beaver
-RUN  chmod a+x /etc/sv/beaver/run && \
-     ln -s /etc/sv/beaver /etc/service/ && \
-     mkdir -p /etc/beaver.d
-
-# Provide a resolver at localhost for software that can't/won't use /etc/resolv.conf
-ADD  docker/service/dnsmasq /etc/service/dnsmasq
-RUN  chmod a+x /etc/service/dnsmasq/run
-
-# disable MTU on vixie cron
-ADD  docker/cron/crontab /etc/crontab
-ADD  docker/cron/root /var/spool/cron/crontabs/root
-RUN  chmod 600 /var/spool/cron/crontabs/root
-ADD  docker/cron/app /var/spool/cron/crontabs/app
-RUN  chown app:crontab /var/spool/cron/crontabs/app && chmod 600 /var/spool/cron/crontabs/app
-
-ADD  docker/my_init.d /etc/my_init.d
-RUN  chmod a+x /etc/my_init.d/*
-
-ADD  docker/confd /etc/confd
-
-RUN mkdir -p /scripts/base
-ADD docker/scripts /scripts/base
-RUN chmod -R a+x /scripts/base
-
-ENV OPG_SERVICE base
+ENV HOME="/root"                     \
+    DEBIAN_FRONTEND="noninteractive" \
+    LANG="en_GB.UTF-8"               \
+    LANGUAGE="en_GB:en"              \
+    LC_ALL="en_GB.UTF-8"             \
+    OPG_SERVICE="base"               \
+    BEAVER_VERSION="36.2.1"          \
+    CONFD_VERSION="0.15.0"           \
+    GOSU_VERSION="1.10"
+
+COPY docker/. /
+
+RUN locale-gen en_GB en_GB.UTF-8 &&\
+    echo "Installing Gosu" &&\
+    gpg --keyserver hkp://ha.pool.sks-keyservers.net --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4 &&\
+    curl -o /usr/local/bin/gosu -SL "https://github.com/tianon/gosu/releases/download/${GOSU_VERSION}/gosu-$(dpkg --print-architecture)" &&\
+    curl -o /usr/local/bin/gosu.asc -SL "https://github.com/tianon/gosu/releases/download/${GOSU_VERSION}/gosu-$(dpkg --print-architecture).asc" &&\
+    gpg --verify /usr/local/bin/gosu.asc &&\
+    rm /usr/local/bin/gosu.asc &&\
+    chmod +x /usr/local/bin/gosu
+
+RUN echo "Updating Packages" &&\
+    apt-get update -qq &&\
+    apt-get install -y git dnsmasq unzip zip python python-pip python-setuptools wget
+
+RUN echo "Installing log aggregation and AWSCLI" &&\
+    pip install -U pip==9.0.3 &&\
+    pip install -q awscli Beaver==${BEAVER_VERSION} &&\
+    mkdir -p /etc/beaver.d &&\
+    chmod a+x /etc/sv/beaver/run  && ln -s /etc/sv/beaver  /etc/service/   &&\
+    chmod a+x /etc/sv/dnsmasq/run && ln -s /etc/sv/dnsmasq /etc/service/
+
+RUN echo "Installing ConfD" &&\
+    curl -o /usr/bin/confd -SL "https://github.com/kelseyhightower/confd/releases/download/v${CONFD_VERSION}/confd-${CONFD_VERSION}-linux-amd64" &&\
+    chmod a+x /usr/bin/confd &&\
+    mkdir -p /etc/confd/conf.d /etc/confd/templates
+
+RUN echo "Disabling ssh access" &&\
+    rm /etc/my_init.d/00_regen_ssh_host_keys.sh &&\
+    rm -rf /etc/service/sshd || true &&\
+    echo "Application user and location" &&\
+    useradd -m -d /app app &&\
```

Comment:
in child images the "app" user is coming through as UID 1000 - but here we're not specifying a UID. 

I'm wondering how reliable this command is without specifying the UID. The UID 1000 compatibility is important with local + CI development, when live volumes are made then IDs need to match on both sides.

Reply:
This is going to be a separate feature request. But very valid. I've made an issue in this repo for it. 
Right now I'm going for squashing layers and leaving things "as is" where at all possible.